src/main/webapp/app/modules/canvas/visualization/util/visualization-render-utils.ts

```diff
@@ -0,0 +1,207 @@
+import $ from 'jquery';
+import flairVisualizations from 'flair-visualizations/js/main';
+import Configuration from 'flair-visualizations/js/extras/configs/configuration';
+import { forwardCall } from 'app/shared/websocket/proxy-websocket.service';
+import { VisualWrap } from './visualmetadata-wrapper';
+const configuration = Configuration();
+
+export const ValidateFields = fields => {
+  let isValid = true;
+  fields.forEach(field => {
+    if (field.feature === null || undefined) {
+      isValid = false;
+      return isValid;
+    }
+  });
+  return isValid;
+};
+
+export const getVisualizationData = (visual, view) => {
+  if (visual.fields && ValidateFields(visual.fields)) {
+    const visualMetadata = VisualWrap(visual);
+    const queryDTO = visualMetadata.getQueryParameters(visual, null, null, null);
+    const body = {
+      queryDTO,
+      visualMetadata,
+      validationType: 'REQUIRED_FIELDS',
+      actionType: null,
+      type: null,
+    };
+    forwardCall(view?.viewDashboard?.dashboardDatasource?.id, body, view.id);
+  }
+};
+
+export const renderVisualization = (visual, metaData, element = 'widget') => {
+  if (visual.fields && ValidateFields(visual.fields)) {
+    const widget = $(`#${element}-${visual.id}`);
+    if (widget.length > 0) {
+      let height = widget[0].clientHeight;
+      const width = widget[0].clientWidth;
+
+      $('#chart-' + visual.id).remove('');
+      if (element === 'widget') {
+        height = height - 30;
+        $('#visualization-' + visual.id).append(
+          '<div id="chart-' +
+            visual.id +
+            '" height="' +
+            height +
+            '" width="' +
+            width +
+            '" style="width:' +
+            width +
+            'px; height:' +
+            height +
+            'px;overflow:hidden;position:relative" ></div>'
+        );
+      } else {
+        $('#visualization-edit-' + visual.id).append(
+          '<div id="chart-' +
+            visual.id +
+            '" height="' +
+            height +
+            '" width="' +
+            width +
+            '" style="width:' +
+            width +
+            'px; height:' +
+            height +
+            'px;overflow:hidden;position:relative" ></div>'
+        );
+      }
```

Comment:
if else condition seems to be generic. only selector is different.try to create a generic code

Reply:
done